includes/options/class-amp-options-manager.php

```diff
@@ -155,6 +175,16 @@ public static function get_options() {
 			$options[ Option::THEME_SUPPORT ] = $defaults[ Option::THEME_SUPPORT ];
 		}
 
+		// Migrate supported post types.
+		if ( isset( $options[ Option::SUPPORTED_POST_TYPES ][0] ) ) {
+			$options[ Option::SUPPORTED_POST_TYPES ] = array_fill_keys( $options[ Option::SUPPORTED_POST_TYPES ], true );
+		}
+
+		// Migrate supported templates.
+		if ( isset( $options[ Option::SUPPORTED_TEMPLATES ][0] ) ) {
+			$options[ Option::SUPPORTED_TEMPLATES ] = array_fill_keys( $options[ Option::SUPPORTED_TEMPLATES ], true );
+		}
```

Comment:
I think this actually needs to take into account the `version`, and make sure the theme args are accounted for here as well.